core/src/main/java/com/linecorp/armeria/server/metric/MetricCollectingServiceBuilder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.metric;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.metric.MeterIdPrefixFunction;
+import com.linecorp.armeria.server.HttpService;
+
+public final class MetricCollectingServiceBuilder {
+
+    private Function<? super RequestLog, Boolean> isSuccess = log -> false;
+    // TODO - what should be the default meter prefix function?
+    private MeterIdPrefixFunction meterIdPrefixFunction = MeterIdPrefixFunction.ofDefault("default");
+
+    MetricCollectingServiceBuilder() { }
+
+    public MetricCollectingServiceBuilder isSuccess(Function<? super RequestLog, Boolean> isSuccess) {
```

Comment:
Add Javadoc and use `Predicate`?
```suggestion
    public MetricCollectingServiceBuilder isSuccess(Predicate<? super RequestLog> successFunction) {
```